x-pack/plugins/observability_solution/investigate_app/server/services/investigation_repository.ts

```diff
@@ -6,10 +6,11 @@
  */
 
 import { Logger, SavedObjectsClientContract } from '@kbn/core/server';
+import { investigationSchema } from '@kbn/investigation-shared/src/schema';
 import { isLeft } from 'fp-ts/lib/Either';
-import { Investigation, StoredInvestigation, investigationSchema } from '../models/investigation';
-import { SO_INVESTIGATION_TYPE } from '../saved_objects/investigation';
+import { Investigation, StoredInvestigation } from '../models/investigation';
```

Comment:
What is the difference between `Investigation` and `StoredInvestigation`?

Reply:
At the moment they are 100% identical, but when using more complex io-ts type, the serialized/deserialized type can be different.
For example, we could have a DateString type that handle DateString <-> Date type, and therefore when serialized (OutputOf), the type is a string (i.e. what we receive on the route, what we send back from the route, and how we store it), but within the backend services, we would use the Date type. Which is much more convenient that having to parse the string every time.

Reply:
Example: https://github.com/kdelemme/kibana/blob/main/x-pack/packages/kbn-slo-schema/src/schema/duration.ts#L13